Computer History Museum Archive.js

```diff
@@ -0,0 +1,589 @@
+				break;
+			case 'Extent':
+				newItem.numPages = getContent(node, fieldTitle).replace(' p.', '')
+				break;
+			case 'Lot Number':
+				newItem.archiveLocation = getContent(node, fieldTitle)
+				break;
+			case 'Description':
+			case 'Type':
+			case 'Subject':
+			case 'Category':
+			case 'Collection Title':
+				newItem.abstractNote = insertToTheStartOfAbstract(getContent(node, fieldTitle), newItem.abstractNote, fieldTitle)
+				break;
+			case 'Identifying Numbers':
+			case 'Copyright Holder':
+			case 'Dimensions':
+			case 'Credit':
+			case 'Place Manufactured':
+			case 'Manufacturer':
+				newItem.extra = addToExtra(newItem.extra, getContent(node, fieldTitle), fieldTitle);
+				break
+		}
+
+	});
+
+
+	let pdfPath = undefined;
+	const pdfDiv = doc.querySelectorAll('div.mediaDocument li a');
+	if (pdfDiv) {
+		pdfDiv.forEach((div, pdfIndex) => {
+			pdfPath = div.href
+			if (pdfPath) {
+				newItem.attachments.push({
+					url: pdfPath.replace('https', 'http'),
+					mimeType: "application/pdf",
+					title: `${newItem.title}${pdfIndex === 0 ? '' : ' ' + pdfIndex + 1}`,
+				});
+			}
+		})
+
+	}
+
+
+	let imagePath = undefined;
+	const imageDiv = doc.querySelectorAll('div.mediarow a.media-large img');
+	if (imageDiv) {
+
+		imageDiv.forEach((div, imageIndex) => {
+			imagePath = div.src
+			if (imagePath) {
+				newItem.attachments.push({
+					url: imagePath,
+					title: `${newItem.title} Image ${imageIndex + 1}`,
+					mimeType: 'image/png'
+				});
+			}
+		})
+
+	}
+
+	let videoPath = undefined;
+	const videoDivs = doc.querySelectorAll('div.mediaVideo iframe')
+	if (videoDivs) {
+		videoDivs.forEach((div, videoIndex) => {
+			videoPath = div.src
+			videoTitle = div.textContent
+			if (videoPath) {
+				newItem.attachments.push({
+					url: videoPath,
+					title: `${newItem.title} video${ videoIndex === 0 ? '' : ' ' + videoIndex + 1}`,
+					snapshot: false
+				});
+			}
+		})
+
+	}
+
+	newItem.complete()
+}
+
+
+// helper functions
+function insertToTheStartOfAbstract(insert, abstract, title) {
+	const abstractContent = title + ': ' + insert + '\n' + (abstract ? abstract : '')
+	return abstractContent  ? abstractContent : ''
+}
+function getContent(node, fieldTitle) {
+	const content = node.textContent.replace(fieldTitle, '').trim()
+	return content ? content : ''
+}
+function addToExtra(oldExtra, newContent, fieldTitle) {
+	return (oldExtra ? oldExtra + '\n' : '') + fieldTitle + ': ' + newContent.replace(/  /g, '').replace(/\n\n/g, '').replace('\n', ': ')
+}
+
+function getContributors(node) {
+	let contributors = []
+	const people = node.querySelectorAll('td')
+	people.forEach(personDiv => {
+		const spans = personDiv.querySelectorAll('span')
+		const name = spans[0].textContent
+		const [lastName, firstName] = name.split(', ')
+		Zotero.debug(lastName)
+		Zotero.debug(firstName)
+		contributors.push({
+			firstName,
+			lastName,
+			creatorType: "contributor",
+			fieldMode: true
```

Comment:
Oh, I see, you're setting this manually. That's not correct. If you have first and last name, you can just omit `fieldMode` (which would be `0` for names with first and last names).

Reply:
fixed. thanks!